packages/webchat/src/main.tsx

```diff
@@ -88,19 +95,30 @@ class Web extends React.Component<MainProps> {
   async load() {
     this.config = this.extractConfig()
 
+    // TODO: 6 - this implies iframe as well. Do we need this? Why?
     if (this.config.exposeStore) {
       const storePath = this.config.chatId ? `${this.config.chatId}.webchat_store` : 'webchat_store'
       set(window.parent, storePath, this.props.store)
     }
 
-    this.config.overrides && this.loadOverrides(this.config.overrides)
+    if (this.config.overrides) {
+      this.loadOverrides(this.config.overrides)
+    }
 
-    this.config.containerWidth && this.postMessageToParent('setWidth', this.config.containerWidth)
+    // TODO: 7 - again with the iframe stuff. Why do we need this.
+    if (this.config.containerWidth) {
+      this.postMessageToParent('setWidth', this.config.containerWidth)
+    }
 
-    this.config.reference && this.props.setReference!()
+    // TODO: 8 - what is reference?
+    if (this.config.reference) {
+      await this.props.setReference!()
+    }
 
+    // TODO: 9 - don't think we need to fetch bot info from a backend. All this can be set at the frontend level
```

Comment:
Aren't we on the frontend? Also we must not forget that the runtime will not be directly reachable, so the frontend cant directly call it to fetch the informations

Reply:
Ideally, we could avoid to put configurations in "window.botpressWebChat.configure" and messaging could provide the configurations from bot.config.json to the frontend directly